pkg/storage/bloom/v1/bloom_tokenizer.go

```diff
@@ -257,14 +226,25 @@ outer:
 
 				// only register the key in the cache if it was successfully added to the bloom
 				// as can prevent us from trying subsequent copies
-				str = string(tok)
-				bt.cache[str] = nil
+				bt.cache[tok] = nil
 				if len(bt.cache) >= cacheSize { // While crude, this has proven efficient in performance testing.  This speaks to the similarity in log lines near each other
 					clear(bt.cache)
 				}
 			}
 		}
 
+		// Only break out of the loop if the bloom filter is full after indexing all structured metadata of an entry.
+		if full {
+			// edge case: one line maxed out the bloom size -- retrying is futile
```

Comment:
Is this comment and _if statement_ relevant now?

IIUC, we will add **at least** one line to the bloom with all it's structured metadata. So the break when full would be sufficient here, no need to skip the next line (I think that's actually buggy).

Reply:
The whole peeking does not make sense any more tbh.